src/tests/components/CardModal/CardModalActions/AttachmentMenu/AttachmentMenu.test.tsx

```diff
@@ -0,0 +1,160 @@
+        value: {
+          id: "1",
+          title: "Test Card",
+          description: "Test Description",
+          listId: "list-id-1",
+          tags: [],
+          attachments: [],
+        },
+      },
+      modalActionMenusVisibility: {
+        value: {
+          ...modalActionsVisibilityState.value,
+          attachmentMenuOpen: {
+            id: 0,
+            status: true,
+          },
+        },
+      },
+    };
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    jest.useRealTimers();
+  });
+
+  it("should render successfully with update true", () => {
+    const view = renderWithProviders(<AttachmentMenu update={true} />, { preloadedState: initialState });
+    expect(view.asFragment()).toMatchSnapshot();
+  });
+
+  it("should render successfully with update false", () => {
+    const view = renderWithProviders(<AttachmentMenu update={false} />, { preloadedState: initialState });
+    expect(view.asFragment()).toMatchSnapshot();
+  });
+
+  it("should close menu", () => {
+    const { store } = renderWithProviders(<AttachmentMenu update={false} />, { preloadedState: initialState });
+
+    const closeIcon = screen.getByTestId("attachment-menu-close") as HTMLElement;
+    expect(closeIcon).toBeInTheDocument();
+
+    const stateBeforeCloseClick = store.getState();
+    expect(stateBeforeCloseClick.modalActionMenusVisibility.value.attachmentMenuOpen.status).toBeTruthy();
+
+    fireEvent.click(closeIcon);
+
+    const stateAfterCloseClick = store.getState();
+    expect(stateAfterCloseClick.modalActionMenusVisibility.value.attachmentMenuOpen.status).toBeFalsy();
+  });
+
+  it("should click the hidden input element", () => {
+    renderWithProviders(<AttachmentMenu update={false} />, { preloadedState: initialState });
+
+    const uploadButton = screen.getByText("Upload from computer") as HTMLButtonElement;
+    const hiddenFileInput = screen.getByTestId("attachment-menu-file-upload") as HTMLInputElement;
+    expect(uploadButton).toBeInTheDocument();
+    expect(hiddenFileInput).toBeInTheDocument();
+
+    jest.spyOn(hiddenFileInput, "click");
+    fireEvent.click(uploadButton);
+
+    expect(hiddenFileInput.click).toHaveBeenCalled();
+  });
+
+  it("should save a single uploaded file successfully", async () => {
+    const { store } = renderWithProviders(<AttachmentMenu update={false} />, { preloadedState: initialState });
+
+    const stateBeforeUpload = store.getState();
+    expect(stateBeforeUpload.currentSelectedCard.value?.attachments).toHaveLength(0);
+    expect(stateBeforeUpload.modalActionMenusVisibility.value.attachmentMenuOpen.status).toBeTruthy();
+
+    const hiddenFileInput = screen.getByTestId("attachment-menu-file-upload") as HTMLInputElement;
+    expect(hiddenFileInput).toBeInTheDocument();
+
+    const uploadedFile = new File(["This is a test file"], "test.txt", { type: "text/plain" });
+    const date = new Date();
+    jest.useFakeTimers().setSystemTime(date);
```

Comment:
This fake timers is needed here as without it, we cannot check the date that is set on the attachment. Using jest to fake the current time ensures that any calls to new Date() will return the `date` variable, meaning we can check the attachment is created with a date correctly and it matches what we expect